### PR TITLE
Refactor `EnumUtils` and make it AOT-friendly.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
### Prerequisites

- [X] I have written a descriptive pull-request title
- [X] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [X] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [ ] ~~I have provided test coverage for my change (where applicable)~~

### Description

The `EnumUtils.Parse` method used to use the non-generic `Enum.GetValues` method to do its job, which is unsafe and produces AOT warnings.
This PR refactors it to use `Enum.IsDefined` instead, which does not allocate.
I also added asserts to methods in `EnumUtils` that ensure we use them with int-sized enums.

### CODE REVIEW

File: src/ImageSharp/Common/Helpers/EnumUtils.cs [hunk 1]
```diff
@@ -19,15 +20,14 @@ internal static class EnumUtils
         /// <param name="defaultValue">The default value to return.</param>
         /// <returns>The <typeparamref name="TEnum"/>.</returns>
         public static TEnum Parse<TEnum>(int value, TEnum defaultValue)
-            where TEnum : Enum
+            where TEnum : struct, Enum
         {
-            foreach (TEnum enumValue in Enum.GetValues(typeof(TEnum)))
+            Debug.Assert(Unsafe.SizeOf<TEnum>() == sizeof(int), "Only int-sized enums are supported.");
```
Comment (antonfirsov) [inline src/ImageSharp/Common/Helpers/EnumUtils.cs:25]:
```suggestion
            DebugGuard.IsTrue(Unsafe.SizeOf<TEnum>() == sizeof(int), "Only int-sized enums are supported.");
```

We don't use `Debug.Assert`.

File: src/ImageSharp/Common/Helpers/EnumUtils.cs [hunk 2]
```diff
@@ -41,8 +41,10 @@ public static TEnum Parse<TEnum>(int value, TEnum defaultValue)
         /// <param name="flag">The flag.</param>
         /// <returns>The <see cref="bool"/>.</returns>
         public static bool HasFlag<TEnum>(TEnum value, TEnum flag)
-          where TEnum : Enum
+          where TEnum : struct, Enum
         {
+            Debug.Assert(Unsafe.SizeOf<TEnum>() == sizeof(int), "Only int-sized enums are supported.");
```
Comment (antonfirsov) [inline src/ImageSharp/Common/Helpers/EnumUtils.cs:46]:
```suggestion
            DebugGuard.IsTrue(Unsafe.SizeOf<TEnum>() == sizeof(int), "Only int-sized enums are supported.");
```

**Review by antonfirsov — COMMENTED** [review]
Thanks!

File: src/ImageSharp/Common/Helpers/EnumUtils.cs (hunk 1 above)
Comment (teo-tsirpanis) [reply src/ImageSharp/Common/Helpers/EnumUtils.cs]: Fixed.

**Comment by JimBobSquarePants** [comment]
Looks great. Thanks!

**Pull request merged**